### PR TITLE
Batching

### PR DESCRIPTION
**Pull request by davija**
Adding the ability to use GMail batching API and other cleanups.

### CODE REVIEW

**Comment by natoverse** [comment]
I'm not a big fan of the * import style, but these are small enough classes it isn't hard to follow. Batch looks reasonable, most everything else is formatting. :shipit:

**Comment by davija** [comment]
Not a fan of that myself either...  I'll have to check my profile to find out, but it almost looks like it is set to switch to import * if >= 3.  This is a really old formatting/config profile so it likely needs to be updated.
